src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -1,58 +1,123 @@
 package com.es.phoneshop.model.product;
 
-import com.es.phoneshop.Exception.ProductNotFoundException;
+import com.es.phoneshop.exception.ProductNotFoundException;
 import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.ExpectedException;
 
 import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Currency;
 
-import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.*;
 
 public class ArrayListProductDaoTest
 {
     private ProductDao productDao;
     private Product testProduct;
 
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
     @Before
     public void setup() {
-        productDao = new ArrayListProductDao();
-        testProduct = new Product(20L, "sgs", "Samsung Galaxy S", new BigDecimal(100), null, 100, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Samsung/Samsung%20Galaxy%20S.jpg");
-        productDao.save(testProduct);
+        productDao = ArrayListProductDao.getInstance();
+    }
+
+    private ArrayList<DateAndPriceBean> startPriceHistory(BigDecimal price) {
```

Comment:
https://medium.com/@sanchezdale/oop-good-practices-coding-to-the-interface-baea84fd60d3